diff --git a/base/compiler/optimize.jl b/base/compiler/optimize.jl
index d83d13831d..bd8f92e298 100644
--- a/base/compiler/optimize.jl
+++ b/base/compiler/optimize.jl
@@ -234,7 +234,7 @@ function optimize(opt::OptimizationState, @nospecialize(result))
     if force_noinline
         opt.src.inlineable = false
     elseif isa(def, Method)
-        if opt.src.inlineable && isdispatchtuple(opt.linfo.specTypes)
+        if opt.src.inlineable# && isdispatchtuple(opt.linfo.specTypes)
             # obey @inline declaration if a dispatch barrier would not help
         else
             bonus = 0